packages/create-amplify/src/logger.ts

```diff
@@ -1,63 +1,138 @@
+  }
 
   /**
    * Logs a message to the console.
    */
-  async log(message: string, level: LogLevel = LogLevel.INFO) {
+  log(message: string, level: LogLevel = LogLevel.INFO) {
     const toLogMessage = level <= this.minimumLogLevel;
 
     if (!toLogMessage) {
       return;
     }
 
-    const logMessage = `[${
-      LogLevel[level]
-    }] ${new Date().toISOString()}: ${message}`;
-    this.console.log(logMessage);
+    const logMessage =
+      this.minimumLogLevel === LogLevel.DEBUG
+        ? `[${LogLevel[level]}] ${new Date().toISOString()}: ${message}`
+        : message;
+    this.stdout.write(logMessage + os.EOL);
+  }
+
+  /**
+   * Start animating ellipsis at the end of a log message.
+   */
+  startAnimatingEllipsis(message: string) {
+    if (!this.isTTY()) {
+      this.log(message, LogLevel.INFO);
+      return;
+    }
+
+    this.animateMessage = message;
+    const frameLength = 4; // number of desired dots - 1
+    let frameCount = 0;
+    this.writeEscapeSequence(EscapeSequence.HIDE_CURSOR);
+    this.stdout.write(this.animateMessage);
+    this.timer = setInterval(() => {
+      this.writeEscapeSequence(EscapeSequence.CLEAR_LINE);
+      this.writeEscapeSequence(EscapeSequence.MOVE_CURSOR_TO_START);
+      this.stdout.write(
+        this.animateMessage + '.'.repeat(++frameCount % frameLength)
+      );
+    }, this.refreshRate);
+  }
+
+  /**
+   * Stops animating ellipsis and replace with a log message.
+   */
+  stopAnimatingEllipsis() {
+    if (!this.isTTY()) {
+      return;
+    }
+
+    clearInterval(this.timer);
+    this.writeEscapeSequence(EscapeSequence.CLEAR_LINE);
+    this.writeEscapeSequence(EscapeSequence.MOVE_CURSOR_TO_START);
+    this.writeEscapeSequence(EscapeSequence.SHOW_CURSOR);
+    this.stdout.write(`${this.animateMessage}...${os.EOL}`);
+  }
+
+  /**
+   * Writes escape sequence to stdout
+   */
+  writeEscapeSequence(action: EscapeSequence) {
+    if (!this.isTTY()) {
+      return;
+    }
+
+    this.stdout.write(action);
+  }
+
+  /**
+   * Checks if the environment is TTY
+   */
+  isTTY() {
+    return this.stdout.isTTY;
   }
 
   /**
    * Logs an error to the console.
    */
-  async error(message: string) {
-    await this.log(message, LogLevel.ERROR);
+  error(message: string) {
+    this.log(message, LogLevel.ERROR);
   }
 
   /**
    * Logs a warning to the console.
    */
-  async warn(message: string) {
-    await this.log(message, LogLevel.WARNING);
+  warn(message: string) {
+    this.log(message, LogLevel.WARNING);
   }
 
   /**
    * Logs an info message to the console.
    */
-  async info(message: string) {
-    await this.log(message, LogLevel.INFO);
+  info(message: string) {
+    this.log(message, LogLevel.INFO);
   }
 
   /**
    * Logs a debug message to the console.
    */
-  async debug(message: string) {
-    await this.log(message, LogLevel.DEBUG);
+  debug(message: string) {
+    this.log(message, LogLevel.DEBUG);
   }
 }
+
+export enum EscapeSequence {
```

Comment:
this shouldn't be exported

Reply:
Done in [2952c9a](https://github.com/aws-amplify/amplify-backend/pull/595/commits/2952c9a94a0c0ed95148bd5a241fa5a50af989ac)